src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -643,173 +657,64 @@ def create_or_upgrade_app(
         policy: PolicyBase,
         interactive: bool,
     ):
-        model = self._entity_model
-        console = self._workspace_ctx.console
-        debug_mode = model.debug
+        sql_executor = get_sql_executor()
 
-        stage_fqn = stage_fqn or package.stage_fqn
-        stage_schema = extract_schema(stage_fqn)
+        event_sharing = EventSharingHandler(
+            telemetry_definition=self.telemetry,
+            deploy_root=package.deploy_root,
+            install_method=install_method,
+            console=self.console,
+        )
 
-        sql_executor = get_sql_executor()
         with sql_executor.use_role(self.role):
-            event_sharing = EventSharingHandler(
-                telemetry_definition=model.telemetry,
-                deploy_root=package.deploy_root,
-                install_method=install_method,
-                console=console,
-            )
 
             # 1. Need to use a warehouse to create an application object
             with sql_executor.use_warehouse(self.warehouse):
 
                 # 2. Check for an existing application by the same name
                 show_app_row = self.get_existing_app_info()
 
+                stage_fqn = stage_fqn or package.stage_fqn
+
                 # 3. If existing application is found, perform a few validations and upgrade the application object.
+                create_or_upgrade_result = None
```

Comment:
if we remove the self._upgrade_app() and _create_app() functions (since they do not do much), would this logic become clearer?

Reply:
I think it's ok as it is now - since create and upgrade do different things, e.g. create needs permission grants and upgrade has specific error handling. Either way I don't have strong feelings about it :D